sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -1952,6 +1952,96 @@ public async Task UpdateSqlServerDataFeedWithEveryMemberAndNewInstance()
             ValidateSqlServerDataSource(updatedDataFeed.DataSource as SqlServerDataFeedSource);
         }
 
+        [RecordedTest]
+        public async Task GetDataFeeds()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            var dataFeedCount = 0;
+
+            await foreach (DataFeed dataFeed in adminClient.GetDataFeedsAsync())
+            {
+                Assert.That(dataFeed.Id, Is.Not.Null.And.Not.Empty);
+                Assert.That(dataFeed.Name, Is.Not.Null.And.Not.Empty);
+                Assert.That(dataFeed.Description, Is.Not.Null);
+                Assert.That(dataFeed.Status, Is.Not.Null);
+                Assert.That(dataFeed.Status, Is.Not.EqualTo(default(DataFeedStatus)));
+                Assert.That(dataFeed.AccessMode, Is.Not.Null);
+                Assert.That(dataFeed.AccessMode, Is.Not.EqualTo(default(DataFeedAccessMode)));
+                Assert.That(dataFeed.ActionLinkTemplate, Is.Not.Null);
+                Assert.That(dataFeed.Creator, Is.Not.Null.And.Not.Empty);
+                Assert.That(dataFeed.Administrators, Is.Not.Null);
+                Assert.That(dataFeed.Viewers, Is.Not.Null);
+                Assert.That(dataFeed.IsAdministrator, Is.Not.Null);
+                Assert.That(dataFeed.CreatedTime, Is.Not.Null);
+                Assert.That(dataFeed.CreatedTime, Is.Not.EqualTo(default(DateTimeOffset)));
+
+                Assert.That(dataFeed.MissingDataPointFillSettings, Is.Not.Null);
+                Assert.That(dataFeed.MissingDataPointFillSettings.FillType, Is.Not.Null);
+                Assert.That(dataFeed.MissingDataPointFillSettings.FillType, Is.Not.EqualTo(default(DataFeedMissingDataPointFillType)));
+
+                if (dataFeed.MissingDataPointFillSettings.FillType == DataFeedMissingDataPointFillType.CustomValue)
+                {
+                    Assert.That(dataFeed.MissingDataPointFillSettings.CustomFillValue, Is.Not.Null);
+                }
+                else
+                {
+                    Assert.That(dataFeed.MissingDataPointFillSettings.CustomFillValue, Is.Null);
+                }
+
+                Assert.That(dataFeed.Granularity, Is.Not.Null);
+                Assert.That(dataFeed.Granularity.GranularityType, Is.Not.EqualTo(default(DataFeedGranularityType)));
```

Comment:
we know for sure that the types that are saved are never going to be assigned the default? (is `Yearly` the default?)

Reply:
It's an extensible enum, so default would be something like:
```cs
var type = new DataFeedGranularityType();
```
which assigns a `null` value to the inner string.

Not exactly what you asked, but we also know that it won't be null because users must specify a granularity type when creating the data feed (and it can't be changed later).